libnavigation-core/src/test/java/com/mapbox/navigation/core/routerefresh/ExpiringDataRemoverTest.kt

```diff
@@ -0,0 +1,192 @@
+package com.mapbox.navigation.core.routerefresh
+
+import com.mapbox.api.directions.v5.models.LegAnnotation
+import com.mapbox.navigation.base.internal.time.parseISO8601DateToLocalTimeOrNull
+import com.mapbox.navigation.core.RouteProgressData
+import com.mapbox.navigation.core.RoutesProgressData
+import com.mapbox.navigation.testing.factories.createDirectionsRoute
+import com.mapbox.navigation.testing.factories.createIncident
+import com.mapbox.navigation.testing.factories.createNavigationRoute
+import com.mapbox.navigation.testing.factories.createRouteLeg
+import io.mockk.every
+import io.mockk.mockk
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import java.util.Date
+
+class ExpiringDataRemoverTest {
+
+    private val localDateProvider = mockk<() -> Date>(relaxed = true)
+    private val sut = ExpiringDataRemover(localDateProvider)
+
+    @Test
+    fun removeExpiringDataFromRoutes() {
+        every {
+            localDateProvider()
+        } returns parseISO8601DateToLocalTimeOrNull("2022-06-30T20:00:00Z")!!
```

Comment:
it's probably more of the matter or preferences though

Reply:
Well, here I set the time it should return in the test (because the test holds the context of what time we should return), while the object is created earlier. Since there is only one test, I can move the object creation into the test, but I'm not sure if it will make a lot of sense. If I add another test in the future, I may want different time.